components/bl-totop-component/src/index.js

```diff
@@ -0,0 +1,58 @@
+const { cn } = BackendlessUI.CSSUtils;
+
+const Transform = {
+  MAIN: "50%",
+  X: "translateX(-50%)",
+  Y: "translateY(-50%)",
+};
+
+export default function TotopComponent({ component }) {
+  const {
+    classList,
+    offset,
+    position,
+    backgroundColor,
+    color,
+    size,
+    iconSize,
+    indent = component.indent ? component.indent : "20px",
+    element,
+  } = component;
+
+  const styles = {
+    backgroundColor,
+    width: size,
+    height: size,
+    top: position.includes("top") ? indent : position.includes("center-") ? Transform.MAIN : undefined,
+    bottom: position.includes("bottom") ? indent : position.includes("center-") ? Transform.MAIN : undefined,
+    right: position.includes("right") ? indent : position.includes("-center") ? Transform.MAIN : undefined,
+    left: position.includes("left") ? indent : position.includes("-center") ? Transform.MAIN : undefined,
+    transform: position.includes("-center") ? Transform.X : position.includes("center-") ? Transform.Y : undefined,
+  };
```

Comment:
How about that approach? 
https://getbootstrap.com/docs/5.0/utilities/position/

Reply:
I think these approaches are roughly equivalent. Applying classes will also require checking the conditions for their application, right there everything you need is immediately collected in styles.